### PR TITLE
Implement wallet withdrawal flow

### PR DESCRIPTION
**Pull request by Artem-Semenov-dev**
This PR implements wallet withdrawal flow. 
The process modeling of wallet withdrawal flow:
image

### CODE REVIEW

File: server/src/test/java/io/spine/examples/shareaware/server/wallet/MoneyCalculatorAbstractTest.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        testNegativeUnits(operation);
+        testNanosOutOfBounds(operation);
+    }
+
+    private static void testBooleanCalculation(Money first, Money second, boolean expected,
+                                               BiFunction<Money, Money, Boolean> operation) {
+        boolean actual = operation.apply(first, second);
+
+        assertThat(actual).isEqualTo(expected);
+        testDifferentCurrencies(operation);
```
Comment (Vladyslav-Kuksiuk) [inline server/src/test/java/io/spine/examples/shareaware/server/wallet/MoneyCalculatorAbstractTest.java:80]: These negative tests call in each test case. Is it supposed to be like this?

File: server/src/test/java/io/spine/examples/shareaware/server/wallet/MoneyCalculatorAbstractTest.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    }
+
+    private static <R> void testDifferentCurrencies(BiFunction<Money, Money, R> operation) {
+        Money first = moneyOf(20, Currency.USD);
+        Money second = moneyOf(30, Currency.UAH);
+
+        IllegalStateException exception =
+                Assertions.assertThrows(IllegalStateException.class,
+                                        () -> operation.apply(first, second));
+        assertThat(exception.getMessage()).
```
Comment (Vladyslav-Kuksiuk) [inline server/src/test/java/io/spine/examples/shareaware/server/wallet/MoneyCalculatorAbstractTest.java:92]: Move `.` to the next line.

**Review by Vladyslav-Kuksiuk — COMMENTED** [review]
@Artem-Semenov-dev please see my comments.

**Comment by armiol** [comment]
@Artem-Semenov-dev I think the captions on the board (at least from what I can see in the screenshot attached) are a bit out of date. Please review.

File: server/src/test/java/io/spine/examples/shareaware/server/wallet/WalletWithdrawalTest.java [hunk 3]
```diff
@@ -0,0 +1,404 @@
...
+            ControllablePaymentGatewayProcess.switchToEventsMode();
+        }
+    }
+
+    @Nested
+    @DisplayName("interact with `PaymentGateway`")
+    class PaymentGateway {
+
+        @Test
+        @DisplayName("which emit the `MoneyTransferredToUser` event")
```
Comment (armiol) [inline server/src/test/java/io/spine/examples/shareaware/server/wallet/WalletWithdrawalTest.java:382]: "emits"?

File: server/src/test/java/io/spine/examples/shareaware/server/wallet/WalletWithdrawalTest.java [hunk 4]
```diff
@@ -0,0 +1,404 @@
...
+    protected BoundedContextBuilder contextBuilder() {
+        return WithdrawalTestContext.newBuilder();
+    }
+
+    @Nested
+    @DisplayName("reduce the wallet balance")
+    class WithdrawMoneyFromBalance {
+
+        @Test
+        @DisplayName("for 400 USD")
```
Comment (armiol) [inline server/src/test/java/io/spine/examples/shareaware/server/wallet/WalletWithdrawalTest.java:80]: How do we know that's $400? It's nowhere near in the code of this test.

In such cases, please don't refer to exact test values in display names. It's difficult to verify (and maintain later).

**Comment by Artem-Semenov-dev** [comment]
> @Artem-Semenov-dev I think the captions on the board (at least from what I can see in the screenshot attached) are a bit out of date. Please review.

I have updated the captions, thank you.

**Review by armiol — APPROVED** [review]
@Artem-Semenov-dev LGTM with some comments to be addressed prior to merging.

**Pull request merged**